packages/fx-core/src/core/FxCore.ts

```diff
@@ -1026,9 +1027,11 @@ export class FxCore implements v3.ICore {
     if (!ctx) return err(new ObjectIsUndefinedError("getProjectConfig input stuff"));
     inputs.stage = Stage.getProjectConfig;
     setCurrentStage(Stage.getProjectConfig);
+    let envState = ctx!.solutionContext?.envInfo.state;
+    if (envState) envState = convertEnvStateMapV3ToV2(envState);
     return ok({
       settings: ctx!.projectSettings,
-      config: ctx!.solutionContext?.envInfo.state,
+      config: envState,
```

Comment:
Maybe need to check and ensure each call of this method uses the right key.

Reply:
FxCore makes sure the exported state and projectSettings are parity with v2.
I will unifty these two methods (getProjectConfig and getProjectConfigV3) in the future.